src/main/java/org/tron/p2p/P2pService.java

```diff
@@ -29,6 +29,8 @@ public void start(P2pConfig p2pConfig) {
     ChannelManager.init();
     DnsManager.init();
     log.info("P2p service started");
+
+    Runtime.getRuntime().addShutdownHook(new Thread(this::close));
```

Comment:
It should be closed manually.
<img width="506" alt="image" src="https://github.com/tronprotocol/libp2p/assets/82020050/62562853-be89-4806-a711-65574d92092d">

Reply:
libp2p can be an independent process, so it uses shutdown hook. Any thread pool in libp2p will not be closed twice as it will read the condition. We may test it more completely after merging it into java-tron.